src/main/groovy/edu/wpi/first/gradlerio/wpi/toolchain/WPIToolchainPlugin.groovy

```diff
@@ -290,12 +290,23 @@ class WPIToolchainPlugin implements Plugin<Project> {
                 if (bin.toolChain in VisualCpp) {
                     bin.cppCompiler.args << '/std:c++14' << '/FS' << '/EHsc' << '/DNOMINMAX'
 
-                    if (bin.buildType.name.equals('debug')) {
-                        bin.cppCompiler.args << '/Zi'
-                        bin.linker.args << '/DEBUG' << '/DEBUG:FULL'
+                    // Both release and debug should enable the the debugging options
+                    // and MT crt mode. Debugging here is always safe
+                    bin.cppCompiler.args << '/Zi' << '/Zc:inline' << '/MT'
+                    bin.linker.args << '/DEBUG:FULL'
+
+                    // Build release in optimized mode.
+                    if (bin.buildType.name.equals('release')) {
+                        bin.cppCompiler.args << '/O2'
```

Comment:
If we ever want to bring PF / the state-space stuff in, `/O2` will need to be the default, I think we should make it apply to all buildTypes.

Reply:
We can make that the case then

Reply:
Fixed